### PR TITLE
Add in card partial scaffolding

### PR DESCRIPTION
**Pull request by carakessler**
This adds in a partial to render a card component in the report, and modifies some existing infrastructure to allow for pdf data to be generated.

### CODE REVIEW

File: packages/cli/__tests__/pdf-test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+describe('generateHTML', () => {
+  it('returns correct HTML string', async () => {
+    const htmlString = await generateHTML(mergedResults);
+
+    expect(htmlString).toMatchSnapshot();
+  });
+  it('includes tailwind UI lib and stylesheet', async () => {
+    const htmlString = await generateHTML(mergedResults);
+
+    expect(htmlString).toContain('tailwind');
```
Comment (scalvert) [inline packages/cli/__tests__/pdf-test.ts]: Should you assert more specifically on tailwind and tailwindui?

File: packages/core/src/types/index.ts [hunk 2]
```diff
@@ -40,10 +40,16 @@ export type JsonTaskResult = {
...
+  stdout = 'stdout',
+  json = 'json',
+  pdf = 'pdf',
+}
+
 export interface TaskResult {
   stdout: () => void;
   json: () => JsonTaskResult;
-  pdf: () => void;
+  pdf: () => void | CardData;
```
Comment (scalvert) [inline packages/core/src/types/index.ts]: This is a bit of a weird intersection type. You'd typically do `CardData | undefined`, not `void`. Can we change this?

File: packages/cli/__tests__/pdf-test.ts (hunk 1 above)
Comment (carakessler) [reply packages/cli/__tests__/pdf-test.ts]: i was waiting to start actually using tailwind to do more testing here... what did you have in mind? we arent actually doing much w tailwind yet

File: packages/core/src/types/index.ts (hunk 2 above)
Comment (carakessler) [reply packages/core/src/types/index.ts]: changed

File: packages/cli/__tests__/pdf-test.ts (hunk 1 above)
Comment (scalvert) [reply packages/cli/__tests__/pdf-test.ts]: I think specifically you're looking at whether those two CSS files are present in the report. This seems good, but I'd use more specific strings to validate this.

```
expect(html).toContain('tailwind.min.css');
expect(html).toContain('tailwind-ui.min.css');
```

Comment (carakessler) [reply packages/cli/__tests__/pdf-test.ts]: fixed :)

File: packages/cli/__tests__/index-test.ts [hunk 3]
```diff
@@ -140,7 +146,7 @@ describe('@checkup/cli', () => {
       await cmd.run(['run', project.baseDir]);
 
       expect(stdout()).toMatchSnapshot();
-    }, 15000);
+    }, 30000);
```
Comment (scalvert) [inline packages/cli/__tests__/index-test.ts]: Should we just make this a constant so we can change it in one place?

Comment (carakessler) [reply packages/cli/__tests__/index-test.ts]: cute ok

File: packages/cli/__tests__/index-test.ts [hunk 4]
```diff
@@ -6,6 +6,8 @@ import { CheckupProject, Plugin, createTmpDir, stdout } from '@checkup/test-help
 
 import cmd = require('../src');
 
+const increasedTestTimeout = 30000;
```
Comment (scalvert) [inline packages/cli/__tests__/index-test.ts]:
```suggestion
const TEST_TIMEOUT = 30000;
```

**Pull request merged**